src/main/java/Duke.java

```diff
@@ -1,10 +1,169 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+  private static final String FILE_PATH = "./data/duke.txt";
+  private static Task[] tasks = new Task[100];
+  private static int taskCount = 0;
+
+  public static void main(String[] args) {
+    // Print the welcome message
+    System.out.println("____________________________________________________________");
+    System.out.println(" Hello! I'm Ruhi.");
+    System.out.println(" What can I do for you?");
+    System.out.println("____________________________________________________________");
+
+    // Load existing tasks from file
+    loadTasksFromFile();
+
+    // Initialize a scanner to read user input
+    Scanner scanner = new Scanner(System.in);
+
+    while (true) {
+      try {
+        String input = scanner.nextLine();
+        System.out.println("____________________________________________________________");
+
+        if (input.equals("bye")) {
+          System.out.println(" Bye. Hope to see you again soon!");
+          System.out.println("____________________________________________________________");
+          break;
+        } else if (input.equals("list")) {
+          // Display the list of tasks
+          System.out.println(" Here are the tasks in your list:");
+          for (int i = 0; i < taskCount; i++) {
+            System.out.println((i + 1) + "." + tasks[i]);
+          }
+          System.out.println("____________________________________________________________");
+        } else if (input.startsWith("mark")) {
+          handleMarkCommand(input);
+        } else if (input.startsWith("unmark")) {
+          handleUnmarkCommand(input);
+        } else if (input.startsWith("delete")) {
+          handleDeleteCommand(input);
+        } else if (input.trim().isEmpty()) {
+          throw new DukeException("Oops! Task description cannot be empty.");
+        } else {
+          handleAddCommand(input);
+        }
+      } catch (DukeException e) {
+        System.out.println(e.getMessage());
+        System.out.println("____________________________________________________________");
+      } catch (Exception e) {
+        System.out.println("Oopsy daisy! Something went wrong, check again.");
+        System.out.println("____________________________________________________________");
+      }
+    }
+ 
+    scanner.close();
+  }
```

Comment:
Consider extracting parts of the code the a separate method to avoid having long methods